lib/go/templates/marketV2_templates.go

```diff
@@ -0,0 +1,387 @@
+package templates
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go-sdk"
+)
+
+// This contains template transactions for the second version of the Top Shot
+// marketplace, which uses a capability to access the owner's moment collection
+// instead of storing the moments in the sale collection directly
+
+// GenerateCreateSaleV2Script creates a cadence transaction that creates a Sale collection
+// and stores in in the callers account published
+func GenerateCreateSaleV2Script(ftAddr, topshotAddr, marketAddr, beneficiaryAddr flow.Address, tokenStorageName string, cutPercentage float64) []byte {
+
+	template := `
+		import FungibleToken from 0x%[5]s
+		import TopShot from 0x%[6]s
+		import TopShotMarketV2 from 0x%[1]s
+
+		transaction {
+			prepare(acct: AuthAccount) {
+				let ownerCapability = acct.getCapability<&{FungibleToken.Receiver}>(/public/%[3]sReceiver)!
+				let beneficiaryCapability = getAccount(0x%[2]s).getCapability<&{FungibleToken.Receiver}>(/public/%[3]sReceiver)!
+
+				let ownerCollection: Capability<&TopShot.Collection> = acct.link<&TopShot.Collection>(/private/MomentCollection, target: /storage/MomentCollection)!
+
+				let collection <- TopShotMarketV2.createSaleCollection(ownerCollection: ownerCollection, ownerCapability: ownerCapability, beneficiaryCapability: beneficiaryCapability, cutPercentage: %[4]f)
+				
+				acct.save(<-collection, to: TopShotMarketV2.marketStoragePath)
+				
+				acct.link<&TopShotMarketV2.SaleCollection{TopShotMarketV2.SalePublic}>(TopShotMarketV2.marketPublicPath, target: TopShotMarketV2.marketStoragePath)
+			}
+		}`
+	return []byte(fmt.Sprintf(template, marketAddr, beneficiaryAddr, tokenStorageName, cutPercentage, ftAddr, topshotAddr))
+}
+
+// GenerateStartSaleV2Script creates a cadence transaction that starts a sale by depositing
+// an NFT into the Sale Collection with an associated price
+func GenerateStartSaleV2Script(topshotAddr, marketAddr flow.Address, id, price int) []byte {
+	template := `
+		import TopShot from 0x%[1]s
+		import TopShotMarketV2 from 0x%[2]s
+
+		transaction {
+			prepare(acct: AuthAccount) {
+
+				let topshotSaleCollection = acct.borrow<&TopShotMarketV2.SaleCollection>(from: TopShotMarketV2.marketStoragePath)
+					?? panic("Could not borrow from sale in storage")
+
+				topshotSaleCollection.listForSale(tokenID: %[3]d, price: %[4]d.0)
+			}
+		}`
+	return []byte(fmt.Sprintf(template, topshotAddr, marketAddr, id, price))
+}
+
+// GenerateCreateAndStartSaleV2Script creates a cadence transaction that creates a Sale collection
+// and stores in in the callers account, and also puts an NFT up for sale in it
+func GenerateCreateAndStartSaleV2Script(ftAddr, topshotAddr, marketAddr, beneficiaryAddr flow.Address, tokenStorageName string, cutPercentage, price float64, tokenID int) []byte {
```

Comment:
It looks like we're using price as an int in v1 and then converting it to UFix64 in the template, is there a reason it switched to a float64 in v2?
https://github.com/dapperlabs/nba-smart-contracts/blob/e6cd787cebb1eb50d04dccdc27844d3bff35511b/lib/go/templates/market_templates.go#L56

Reply:
It might have been what I copied vs what you actually used in your app. Fee free to change it to whatever you want it to be @DavidRS91